src/huggingface_hub/commands/lfs.py

```diff
@@ -162,33 +171,88 @@ def run(self):
                 }
             )
 
+            # caculate workload per thread
+            n_url = len(presigned_urls)
+            n_thread = min(n_url, ref_concurrency)
+            n_heavy = n_url % n_thread
+            # n_light = n_thread - n_heavy
+            n_per_light_thread = n_url // n_thread
+            n_per_heavy_thread = n_per_light_thread + 1
```

Comment:
Could you explain what you are doing with the heavy/light distinction, I'm not sure I understand what your intent is

Reply:
I think I understand, correct if I'm wrong, you start `n_thread`s concurrent futures within which you do for loops that will send `chunk_size` parts of either a heavy or light part of the file

Reply:
yes, as  chunks may not assigned evenly for each thread, and I want to keep the transfer order in each thread, so the first `n_per_heavy_thread`  thread take more chunks, and the rest take less.